src/tabrmd.c

```diff
@@ -265,6 +265,58 @@ show_version (const gchar  *option_name,
     g_print ("tpm2-abrmd version %s\n", VERSION);
     exit (0);
 }
+/*
+ * Break the 'combined_conf' string up into the name of the TCTI and the
+ * config string passed to the TCTI. The combined_conf string is formatted:
+ * "name:conf" where:
+ * - 'name' is the name of the TCTI library in the form 'libtcti-name.so'.
+ *   Additionally the prefix 'libtcti-' and suffix '.so' may be omitted.
+ * - 'conf' is the configuration string passed to the TCTI. This is TCTI
+ *   specific.
+ * Bot the 'name' and 'conf' fields in this string are optional HOWEVER the
+ * if no semicolon is present in the combined_conf string it will be assumed
+ * that the whole string is the 'name'. To provide a 'conf' string to the
+ * default TCTI the first character of the combined_conf string must be a
+ * semicolon.
+ * If a field in the combined_conf string indicates a default value then
+ * the provided tcti_filename or tcti_conf will not be set. This is to allow
+ * defaults to be set by the caller and only updated here if we're changing
+ * them.
+ */
+int
+tcti_conf_parse (gchar *combined_conf,
+                 gchar **tcti_filename,
+                 gchar **tcti_conf)
+{
+    gchar *split;
+
+    g_debug ("%s", __func__);
+    if (tcti_filename == NULL || tcti_conf == NULL) {
+        g_info ("%s: tcti_filename and tcti_conf out params may not be NULL",
+                __func__);
+        return -1;
+    }
+    if (combined_conf == NULL) {
+        g_debug ("%s: combined conf is null", __func__);
+        return 0;
+    }
+
+    split = strchr (combined_conf, ':');
+    /* no semicolon, combined_conf is tcti name */
+    if (split == NULL) {
+        *tcti_filename = combined_conf;
+        return 0;
+    }
+    split [0] = '\0';
+    if (combined_conf[0] != '\0') {
```

Comment:
I think this is dead code. if combined_conf is the empty string, wont line ```strchr()``` on line 304 return NULL and then bail on line 308?

Reply:
That's what will happen if `combined_conf` is the empty string. But if `combined_conf` is `:` setting `split[0] = '\0'` will cause `combined_conf[0]` to be `'\0` and this conditional will ensure that we don't set `tcti_filename` thus preserving the default.